### ISSUE TITLE
ESP32 - machine.reset() IllegalInstruction

### ISSUE THREAD

**Issue by vshymanskyy** [micropython/micropython#3855]
When calling `machine.reset()` on ESP32, I get:
```
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
Guru Meditation Error: Core  0 panic'ed (IllegalInstruction)
. Exception was unhandled.
...
```
The board indeed resets after this ^_^ but I thing the expected behavior is different!

^D works fine...

**Comment by dpgeorge** [comment]
The same issue was recently fixed in #3810.  Did you use the latest firmware?  Maybe the bug resurfaced...

**Comment by vshymanskyy** [comment]
I'll reopen if it appears on the latest. Provably a duplucate
